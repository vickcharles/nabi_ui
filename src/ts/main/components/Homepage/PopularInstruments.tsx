import * as React from 'react';
import InstrumentCard, { InstrumentCardType } from './InstrumentCard';
import { popularInstruments } from '../../../../assets/data/popularInstruments';
import PageTitle from '../PageTitle';
import GridList, { GridListTile } from 'material-ui/GridList';

/**
 * PopularInstruments: This is the Polular Instruments section of the Homepage
 */

class PopularInstruments extends React.Component<{}, {}> {
  componentDidMount() {
    const slides = document.querySelectorAll(`.popular-instrument-card-group`);
    let currentSlide = 0;
    const nextSlide = () =>  {
        slides[currentSlide].className = 'popular-instrument-card-group';
        currentSlide = (currentSlide + 1) % 3;
        slides[currentSlide].className = 'popular-instrument-card-group nabi-showing';
    };
    setInterval(nextSlide, 4000);
  }

  public render() {
    const popularInstrumentsFirstTrio = popularInstruments.slice(0, 3);
    const popularInstrumentsSecondTrio = popularInstruments.slice(3, 6);
    const popularInstrumentsThirdTrio = popularInstruments.slice(6, 9);

    const renderfirstTrio = (array: InstrumentCardType[]): JSX.Element => {
      return (
        <GridList className="" cellHeight={330} cols={3}>
        {array.map((items, i ) => (
          <GridListTile key={i} cols={1}>
            <InstrumentCard
              key={i}
              index={i}
              image={items.image}
              instrument={items.instrument}
              instructors={items.instructors}
              className=""
            />
          </GridListTile>
          )
        )}
        </GridList>
      );
    };

    const renderInstrumentCard = () => {
      const Cards = popularInstruments.map((items, i ) => {
        return (
          <InstrumentCard
            key={i}
            index={i}
            image={items.image}
            instrument={items.instrument}
            instructors={items.instructors}
            className="popular-instrument-card"
          />
        );
      });
      return Cards;
    };

    return (
      <div className="nabi-margin-top-xlarge nabi-margin-bottom-medium nabi-background-color">
        <PageTitle pageTitle="Popular Instruments" />
        <div className="nabi-container">
          <div className="hide-on-desktop nabi-popular-instruments">
            {renderInstrumentCard()}
          </div>
          <div className="hide-on-mobile nabi-popular-instruments">
          <div className="popular-instrument-card-group nabi-showing">
              {renderfirstTrio(popularInstrumentsFirstTrio)}
            </div>
            <div className="popular-instrument-card-group">
              {renderfirstTrio(popularInstrumentsSecondTrio)}
            </div>
            <div className="popular-instrument-card-group">
              {renderfirstTrio(popularInstrumentsThirdTrio)}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default PopularInstruments;