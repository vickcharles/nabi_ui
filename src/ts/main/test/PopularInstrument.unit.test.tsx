import * as React from 'react';
import PopularInstruments from '../components/Homepage/PopularInstruments';
import { shallow } from 'enzyme';

describe('PopularInstruments', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <PopularInstruments/>
    );
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
