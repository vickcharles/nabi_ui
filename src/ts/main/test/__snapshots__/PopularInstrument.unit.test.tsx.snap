// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularInstruments Renders the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopularInstruments />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PageTitle
          pageTitle="Popular Instruments"
        />,
        <div
          className="nabi-container"
        >
          <div
            className="nabi-popular-instruments"
          >
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={0}
              instructors={27}
              instrument="piano"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-guitar.png"
              index={1}
              instructors={24}
              instrument="guitar"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-drums.jpg"
              index={2}
              instructors={19}
              instrument="drums"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-violin.jpg"
              index={3}
              instructors={18}
              instrument="violin"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={4}
              instructors={15}
              instrument="saxophone"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={5}
              instructors={10}
              instrument="flute"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={6}
              instructors={13}
              instrument="clarinet"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={7}
              instructors={23}
              instrument="electric guitar"
            />
          </div>
        </div>,
      ],
      "className": "nabi-margin-top-xlarge nabi-margin-bottom-medium nabi-background-color",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pageTitle": "Popular Instruments",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="nabi-popular-instruments"
          >
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={0}
              instructors={27}
              instrument="piano"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-guitar.png"
              index={1}
              instructors={24}
              instrument="guitar"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-drums.jpg"
              index={2}
              instructors={19}
              instrument="drums"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-violin.jpg"
              index={3}
              instructors={18}
              instrument="violin"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={4}
              instructors={15}
              instrument="saxophone"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={5}
              instructors={10}
              instrument="flute"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={6}
              instructors={13}
              instrument="clarinet"
            />
            <WithStyles(InstrumentCard)
              image="nabi-home-popular-instruments-piano.png"
              index={7}
              instructors={23}
              instrument="electric guitar"
            />
          </div>,
          "className": "nabi-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={0}
                instructors={27}
                instrument="piano"
              />,
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-guitar.png"
                index={1}
                instructors={24}
                instrument="guitar"
              />,
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-drums.jpg"
                index={2}
                instructors={19}
                instrument="drums"
              />,
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-violin.jpg"
                index={3}
                instructors={18}
                instrument="violin"
              />,
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={4}
                instructors={15}
                instrument="saxophone"
              />,
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={5}
                instructors={10}
                instrument="flute"
              />,
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={6}
                instructors={13}
                instrument="clarinet"
              />,
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={7}
                instructors={23}
                instrument="electric guitar"
              />,
            ],
            "className": "nabi-popular-instruments",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-piano.png",
                "index": 0,
                "instructors": 27,
                "instrument": "piano",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-guitar.png",
                "index": 1,
                "instructors": 24,
                "instrument": "guitar",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-drums.jpg",
                "index": 2,
                "instructors": 19,
                "instrument": "drums",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-violin.jpg",
                "index": 3,
                "instructors": 18,
                "instrument": "violin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-piano.png",
                "index": 4,
                "instructors": 15,
                "instrument": "saxophone",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-piano.png",
                "index": 5,
                "instructors": 10,
                "instrument": "flute",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-piano.png",
                "index": 6,
                "instructors": 13,
                "instrument": "clarinet",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "class",
              "props": Object {
                "image": "nabi-home-popular-instruments-piano.png",
                "index": 7,
                "instructors": 23,
                "instrument": "electric guitar",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PageTitle
            pageTitle="Popular Instruments"
          />,
          <div
            className="nabi-container"
          >
            <div
              className="nabi-popular-instruments"
            >
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={0}
                instructors={27}
                instrument="piano"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-guitar.png"
                index={1}
                instructors={24}
                instrument="guitar"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-drums.jpg"
                index={2}
                instructors={19}
                instrument="drums"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-violin.jpg"
                index={3}
                instructors={18}
                instrument="violin"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={4}
                instructors={15}
                instrument="saxophone"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={5}
                instructors={10}
                instrument="flute"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={6}
                instructors={13}
                instrument="clarinet"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={7}
                instructors={23}
                instrument="electric guitar"
              />
            </div>
          </div>,
        ],
        "className": "nabi-margin-top-xlarge nabi-margin-bottom-medium nabi-background-color",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pageTitle": "Popular Instruments",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="nabi-popular-instruments"
            >
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={0}
                instructors={27}
                instrument="piano"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-guitar.png"
                index={1}
                instructors={24}
                instrument="guitar"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-drums.jpg"
                index={2}
                instructors={19}
                instrument="drums"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-violin.jpg"
                index={3}
                instructors={18}
                instrument="violin"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={4}
                instructors={15}
                instrument="saxophone"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={5}
                instructors={10}
                instrument="flute"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={6}
                instructors={13}
                instrument="clarinet"
              />
              <WithStyles(InstrumentCard)
                image="nabi-home-popular-instruments-piano.png"
                index={7}
                instructors={23}
                instrument="electric guitar"
              />
            </div>,
            "className": "nabi-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-piano.png"
                  index={0}
                  instructors={27}
                  instrument="piano"
                />,
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-guitar.png"
                  index={1}
                  instructors={24}
                  instrument="guitar"
                />,
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-drums.jpg"
                  index={2}
                  instructors={19}
                  instrument="drums"
                />,
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-violin.jpg"
                  index={3}
                  instructors={18}
                  instrument="violin"
                />,
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-piano.png"
                  index={4}
                  instructors={15}
                  instrument="saxophone"
                />,
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-piano.png"
                  index={5}
                  instructors={10}
                  instrument="flute"
                />,
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-piano.png"
                  index={6}
                  instructors={13}
                  instrument="clarinet"
                />,
                <WithStyles(InstrumentCard)
                  image="nabi-home-popular-instruments-piano.png"
                  index={7}
                  instructors={23}
                  instrument="electric guitar"
                />,
              ],
              "className": "nabi-popular-instruments",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-piano.png",
                  "index": 0,
                  "instructors": 27,
                  "instrument": "piano",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-guitar.png",
                  "index": 1,
                  "instructors": 24,
                  "instrument": "guitar",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-drums.jpg",
                  "index": 2,
                  "instructors": 19,
                  "instrument": "drums",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-violin.jpg",
                  "index": 3,
                  "instructors": 18,
                  "instrument": "violin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-piano.png",
                  "index": 4,
                  "instructors": 15,
                  "instrument": "saxophone",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-piano.png",
                  "index": 5,
                  "instructors": 10,
                  "instrument": "flute",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-piano.png",
                  "index": 6,
                  "instructors": 13,
                  "instrument": "clarinet",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "7",
                "nodeType": "class",
                "props": Object {
                  "image": "nabi-home-popular-instruments-piano.png",
                  "index": 7,
                  "instructors": 23,
                  "instrument": "electric guitar",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
